### PR TITLE
feat(filter): Added --metric_filter_path

### PR DESCRIPTION
**Pull request by ewiseblatt**
Deprecated experimental --prototype_path, which is unlikely to be in use

@jtk54

### CODE REVIEW

File: spinnaker-monitoring-daemon/spinnaker-monitoring/metric_filter.py [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+    # If includeTags is present, then exclude all tags but those mentioned.
+    # If excludeTags is present, then exclude the mentioned tags.
+    # These are optional and independent of one another.
+    <explicit metric name>:
+       includeTags: <list>
+       excludeTags: <list>
+
+  # If metric was not explicitly enumerated in "explicit" dict, then
+  # check the metrics metadata against the following lists of regexes
+  - includeNameRegex: <value>
```
Comment (jtk54) [inline spinnaker-monitoring-daemon/spinnaker-monitoring/metric_filter.py]: The `-` marks this as a list entry under `explicit`. Is this the structure, or is `includeNameRegex` another distinct entry under `explicit` or `meters`?

File: spinnaker-monitoring-daemon/spinnaker-monitoring/metric_filter.py [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+    # The name matches the given regex
+
+  - excludeNameRegex: <value>
```
Comment (jtk54) [inline spinnaker-monitoring-daemon/spinnaker-monitoring/metric_filter.py]: Same here.

File: spinnaker-monitoring-daemon/spinnaker-monitoring/metric_filter.py [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+    """
+    metrics = data['values']
+    tags_hash_to_values = {}
+    hash_to_tags = {}
+    for metric in metrics:
+      values = metric['values']
+      tags = sorted([tag for tag in metric['tags']
+                     if tag_func(tag['key'])],
+                    key=lambda entry: entry['key'])
+      hash = str(tags)
```
Comment (jtk54) [inline spinnaker-monitoring-daemon/spinnaker-monitoring/metric_filter.py]: Might not matter too much, but `hash` is a python builtin: https://docs.python.org/2/library/functions.html.

**Comment by ewiseblatt** [comment]
Sorry, I just changed the data model and rewrote the PR in response to one of your observations.
I realized that I messed up the YAML syntax which made my data model was awkward to use. I didnt rebase this comment yet,  though most lines are different in case it helps guide a review.

Before I had explicit meter declarations (along with tags) or regex matching (or excluding) for meters without a way to specify the tags for them. Then globally I had a way to specify tags to include or exclude on those, but not scoped to meters. This was awkward and limited in expression.

The change is you can either specify meters explicity as before, or with regular expressions and tags (or excluding names as before). Literal and regular expression are essentially the same except that literal has higher precedence and is explicit whereas a name satisfied by multiple regex is ambiguous and cannot guarantee which tag policy will follow.

There is still the include/exclude on tags which can have global affect, though explicit tag policy in the meters section will override it.

This is more expressive than before, and I think covers anything you could want from a schema perspective. It might still be tedious to specify, but is more concise than before and not sure how to improve.

**Pull request merged**